src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -629,6 +659,31 @@ public void invalidate(boolean expectMBean)
         strategyContainer.shutdown();
         SystemKeyspace.removeTruncationRecord(metadata.id);
 
+        storageHandler.withReloadingDisabled(() -> {
+            if (dropData)
+            {
+                data.dropSSTables();
+                LifecycleTransaction.waitForDeletions();
+
+                indexManager.dropAllIndexes();
+            }
+            else
+            {
+                // In CNDB, because of multi-tenancy, we might just unload a CFS without deleting the data as
+                // a tenant can be moved to a different set of nodes, which will then need to read data from remote storage
+                data.unloadSSTables();
+
+                indexManager.unloadAllIndexes();
+            }
+
+            storageHandler.unload();
+
+            if (dropData)
+            {
+                LifecycleTransaction.waitForDeletions(); // just in case an index had a reference on the sstable
+            }
+        });
+
         data.dropSSTables();
         LifecycleTransaction.waitForDeletions();
         indexManager.dropAllIndexes();
```

Comment:
Are these original calls to `data.dropSSTables()`, etc supposed to still be around after this change? Aren't we potentially calling them twice now? Is that OK?

Reply:
good catch, I should have removed them